File: LeetCode/Java Solutions/Average Waiting Time/average.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        int currentTime = 0;
        long totalWaitingTime = 0;
        
        for (int[] customer : customers) {
            int arrival = customer[0];
            int time = customer[1];
            currentTime = Math.max(currentTime, arrival) + time;
            totalWaitingTime += currentTime - arrival;
        }
        
        return (double) totalWaitingTime / customers.length;
    }
}

File: LeetCode/Python Solutions/Average Waiting Time/average.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        currentTime = 0
        totalWaitingTime = 0
        
        for arrival, time in customers:
            currentTime = max(currentTime, arrival) + time
            totalWaitingTime += currentTime - arrival
        
        return totalWaitingTime / len(customers)
